packages/compass-aggregations/src/modules/pipeline-builder/stage.ts

```diff
@@ -0,0 +1,128 @@
+import * as babelParser from '@babel/parser';
+import type * as t from '@babel/types';
+import mongodbQueryParser from 'mongodb-query-parser';
+
+import {
+  isNodeDisabled,
+  setNodeDisabled,
+  assertStageNode,
+  getStageValueFromNode,
+  getStageOperatorFromNode,
+  isStageLike,
+} from './pipeline-parser/stage-parser';
+
+const PARSE_ERROR = 'Stage must be a properly formatted document.';
+
+function createStage({
+  key,
+  value
+}: { key?: string; value?: string } = {}): t.ObjectExpression {
+  return {
+    type: 'ObjectExpression',
+    properties: [
+      {
+        type: 'ObjectProperty',
+        ...(key && { key: { type: 'Identifier', name: key } }),
+        ...(value && { value: babelParser.parseExpression(value) }),
+        computed: false,
+        shorthand: false
+        // NB: This is not a completely valid object property: it might be
+        // missing either `key` or `value`, but for our purposes this is
+        // alright as @babel/generator can handle these values missing, so
+        // converting between text pipeline and stages pipeline will be still
+        // possible
+      } as t.ObjectProperty
+    ]
+  };
+}
+
+function assertStageValue(value: string) {
+  // mongodbQueryParser will either throw or return an
+  // empty string if input is not a valid query
+  const parsed = (mongodbQueryParser as any)(value);
+  if (parsed === '') {
+    throw new Error(PARSE_ERROR);
+  }
+}
+
+export function stageToString(operator: string, value: string, disabled: boolean): string {
+  const str = `{
+  ${operator ?? ''}: ${value ?? ''}
```

Comment:
Why do you need nullish here if the type is strictly string bot for operator and value?